sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample09_Extensibility.md

```diff
@@ -0,0 +1,395 @@
+## Extensibility
+
+This sample demonstrates how the key Service Bus types can be extended to provide custom functionality. As an example, we will demonstrate how messages can be mutated before sending and after receiving. This mimics the functionality enabled by the `RegisterPlugin` method on client types in `Microsoft.Azure.ServiceBus`.
```

Comment:
```suggestion
This sample demonstrates how the key Service Bus types can be extended to provide custom functionality. As an example, we will demonstrate how messages can be intercepted and enriched before sending and after receiving. This mimics the functionality enabled by the `RegisterPlugin` method on client types in `Microsoft.Azure.ServiceBus`.
```

Reply:
Suggesting "enrichment" as a common term for the concept in the messaging and event sourcing domain.  
_([example: google search](https://www.google.com/search?q=message+enrichment))_   
_([example: bing search](https://www.bing.com/search?q=message+enrichment))_

Reply:
I considered enrichment, but thought it was slightly inaccurate when thinking about the blob attachment use case.

Reply:
I'm fine with using it if it is the industry standard though.